cohort-util/src/main/java/com/ibm/cohort/valueset/ValueSetUtil.java

```diff
@@ -97,30 +114,43 @@ else if (inCodesSection) {
 		valueSet.setCompose(compose);
 		ValueSetArtifact artifact = new ValueSetArtifact();
 		artifact.setName(valueSet.getName());
-		artifact.setResource(valueSet);
+		artifact.setFhirResource(valueSet);
 		artifact.setUrl(valueSet.getUrl());
 		return artifact;
 	}
 
-	public static void importArtifacts(IGenericClient client, List<ValueSetArtifact> valueSetArtifacts) {
-		importArtifacts(client, valueSetArtifacts, false);
+	public static void importArtifacts(IGenericClient client, List<ValueSetArtifact> valueSetArtifacts, boolean updateIfExists) {
+		for(ValueSetArtifact valueSetArtifact : valueSetArtifacts) {
+			importArtifact(client, valueSetArtifact, updateIfExists);
+		}
+	}
+
+	public static void deleteValueSet(IGenericClient client, String url){
+		client.delete().resourceConditionalByType("ValueSet").where(ValueSet.URL.matches().value(url)).execute();
 	}
 
-	public static void importArtifacts(IGenericClient client, List<ValueSetArtifact> valueSetArtifacts, boolean continueIFExists) {
+	public static String importArtifact(IGenericClient client, ValueSetArtifact valueSetArtifact, boolean updateIfExists) {
 
-		for (ValueSetArtifact valueSetArtifact : valueSetArtifacts) {
-			Bundle bundle = client.search().forResource(ValueSet.class).where(ValueSet.URL.matches().value(valueSetArtifact.getUrl()))
-					.returnBundle(Bundle.class).execute();
-			if (bundle.getEntry().size() > 0) {
-				valueSetArtifact.setId(bundle.getEntryFirstRep().getResource().getIdElement().getIdPart());
-				if (continueIFExists)  {
-					MethodOutcome outcome = client.update().resource(client.getFhirContext().newJsonParser().encodeResourceToString(valueSetArtifact.getResource())).execute();
-					valueSetArtifact.setId(outcome.getId().getIdPart());
-				}
-			} else {
-				MethodOutcome outcome = client.create().resource(client.getFhirContext().newJsonParser().encodeResourceToString(valueSetArtifact.getResource())).execute();
-				valueSetArtifact.setId(outcome.getId().getIdPart());
+		Bundle bundle = client.search().forResource(ValueSet.class).where(ValueSet.URL.matches().value(valueSetArtifact.getUrl()))
+				.returnBundle(Bundle.class).execute();
+		MethodOutcome outcome;
+		if(bundle.getEntry().size() > 0){
+			String[] url = bundle.getEntry().get(0).getFullUrl().split("/");
```

Comment:
Sanity checking myself:
Bundle entries have full url and the actual resource entry. The last part of the full url will always match the resource's id. Is that right?

Reply:
Yup!